core/src/main/java/org/infinispan/remoting/transport/jgroups/JGroupsTracedTransport.java

```diff
@@ -0,0 +1,133 @@
+package org.infinispan.remoting.transport.jgroups;
+
+import java.util.Collection;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.TimeUnit;
+import java.util.function.BiConsumer;
+
+import org.infinispan.commands.ReplicableCommand;
+import org.infinispan.commands.remote.SingleRpcCommand;
+import org.infinispan.configuration.cache.Configuration;
+import org.infinispan.factories.annotations.Inject;
+import org.infinispan.factories.scopes.Scope;
+import org.infinispan.factories.scopes.Scopes;
+import org.infinispan.manager.EmbeddedCacheManager;
+import org.infinispan.registry.InternalCacheRegistry;
+import org.infinispan.remoting.inboundhandler.DeliverOrder;
+import org.infinispan.remoting.transport.Address;
+import org.infinispan.remoting.transport.ResponseCollector;
+import org.infinispan.security.actions.SecurityActions;
+import org.infinispan.telemetry.InfinispanSpan;
+import org.infinispan.telemetry.InfinispanSpanAttributes;
+import org.infinispan.telemetry.InfinispanTelemetry;
+import org.infinispan.telemetry.SpanCategory;
+import org.infinispan.telemetry.impl.DisabledInfinispanSpan;
+import org.infinispan.xsite.SingleXSiteRpcCommand;
+
+@Scope(Scopes.GLOBAL)
+public class JGroupsTracedTransport extends JGroupsTransport {
```

Comment:
Trying to introduce the factory, IIUC it will be abstract factory pattern here.

Reply:
I think I prefer to start with a single class factory (factory method instead of abstract factory)... in case we can evolve it.

Reply:
@wburns I applied a different (probably a simpler idea). Instead of touching the way the requests are created (introducing factory), I simply introduced a tracing method on a dedicated object (composition instead of extension). I don't know what you think about it. IMO it looks quite compact and safe.